pandas/core/dtypes/base.py

```diff
@@ -422,28 +427,52 @@ def register(self, dtype: type[ExtensionDtype]) -> None:
 
         self.dtypes.append(dtype)
 
-    def find(self, dtype: type[ExtensionDtype] | str) -> type[ExtensionDtype] | None:
+    @overload
+    def find(self, dtype: type[ExtensionDtypeT]) -> ExtensionDtypeT:
```

Comment:
```
>>> typ
<class 'pandas.core.arrays.integer.Int64Dtype'>
>>> pd.core.dtypes.base.Registry().find(typ)
<class 'pandas.core.arrays.integer.Int64Dtype'>
>>> 
```
```suggestion
    def find(self, dtype: type[ExtensionDtypeT]) -> type[ExtensionDtypeT]:
```

Reply:
Turns out that `type[ExtensionDtypeT]` is not a possible return type, so I removed it.

Reply:
see code sample in comment

Reply:
Right, so I can put it back, but then `type[ExtensionDtypeT]` conflicts with `npt.DTypeLike`, because that can be any type, so have to go back to specifying the specific types.

Reply:
type[object] can create issues. but unfortunately `object` is a valid dtype, use ignores where needed.

Reply:
> type[object] can create issues. but unfortunately `object` is a valid dtype, use ignores where needed.

So the issue here is that if I use `npt.DTypeLike`, then the signatures with `type_t[ExtensionDtypeT]` and `npt.DTypeLike` overlap, because `npt.DTypeLike` includes `type` in the `Union`.  Are you suggesting to use `npt.DtypeLike` instead of  `NpDtype | type_t[str | float | int | complex | bool | object]` and use `ignore` on the overload overlapping?  Doesn't seem right to me.

Reply:
The goal here is to avoid false positives for users passing a valid dtype and also avoid using our `NpDtype` alias. Ideally we want to remove use of that alias completely.

Examining this function and ignoring the docstring, any type can be passed and the return would be `None` if not an EA type. i.e. this function never raises a TypeError.

maybe widen the types to any, use `object` and if mypy complains use `Any` or ignore?

Reply:
If working with the public api, you may want to consider reviving #40202. 

if not create a test file and post the results (and the test file used) in a similar maner to #40200

Reply:
`Registry.find()` is not in the public API, so do we need to worry about "false positives for users" ?  This code is just so pandas developers don't mess things up.

Reply:
ah yes, The registry attribute was privatized in #40538 so that `pyright --verifytypes` passes. The changes in this PR are not to fix pyright issues?

Reply:
TBH, I don't remember!  It's been since January since I started this adventure.  I did this PR as part of getting `ExtensionArray` working.  I haven't been using `pyright` to verify this.